en/flight_controller/snapdragon_flight_software_installation.md

```diff
@@ -0,0 +1,215 @@
+| MAVROS	| 0.19.0				|[399a04e](https://github.com/nicovanduijn/mavros/commit/399a04ef5739bd93fe1c596b63d6bdbfba7d84ef)|
+| MAVLink	| release/kinetic/mavlink/2017.5.5-0 	|[2ec76d2](https://github.com/mavlink/mavlink-gbp-release/commit/2ec76d25c6b7e6fac1f6b0d08765d07aa0f4f69d)|
+| PX4		| 1.7.3 + additions			|[087f0f8](https://github.com/ChristophTobler/Firmware/commit/087f0f8bf014a6b073c12408f35cb9fb2e588af9)|
+| Snap VIO	| mv-1.0.2/mavros-integration		|[eecc3c2](https://github.com/nicovanduijn/snap-vislam-ros/commit/eecc3c24cb2dc4451df1e97f699fffdaac04f266)|
+| Snapdragon-Linux| 3.1.3.1				|[here](https://support.intrinsyc.com/projects/snapdragon-flight/files)|
+| DSP Firmware	| 3.1.3.1				|[here](https://support.intrinsyc.com/projects/snapdragon-flight/files)|
+| Qualcomm MV	| 1.0.2					|[here](https://developer.qualcomm.com/sdflight-tools)|
+
+
+## Prevent bricking
+To prevent the system from hanging on boot because of anything wrong with the ADSP firmware, do the following changes before going any further.
+Plug your snapdragon into your computer via USB and open the android debug shell:
+`adb shell`
+
+Edit the file `/usr/local/qr-linux/q6-admin.sh`:
+```
+vim /usr/local/qr-linux/q6-admin.sh
+```
+
+Comment out the following lines:
+```
+# Wait for adsp.mdt to show up
+	#while [ ! -s /lib/firmware/adsp.mdt ]; do
+	#  sleep 0.1
+	#done
+```
+
+
+And:
+```
+# Don't leave until ADSP is up
+	#while [ "`cat /sys/kernel/debug/msm_subsys/adsp`" != "2" ]; do
+	#  sleep 0.1
+	#done
+```
+
+## Update Linux Image
+> **Warning** Updating the Linux Image on your snapdragon will erase everything.
+
+Get the latest `Flight_x.x_JFlash.zip` from [here](https://support.intrinsyc.com/projects/snapdragon-flight/files) and unzip it. In the unzipped folder is a script that has to be used to update the Linux image. Power your Snapdragon Flight, connect it using a micro USB cable and run
+
+`sudo ./jflash.sh`
+
+
+## Update DSP processor firmware
+Get the latest `Flight_x.x_qcom_flight_controller_hexagon_sdk_add_on.zip` from [here](https://support.intrinsyc.com/projects/snapdragon-flight/files) and unzip it. In the unzipped folder, run
+```
+./installfcaddon.sh
+adb shell
+sudo reboot
+```
+
+## Clone PX4 Firmware & build
+On your PC, clone the PX4 firmware repo and build it like so:
+
+If you haven't yet cloned the Firmware repo:
+```
+cd ~
+mkdir src
+cd src
+git clone git@github.com:ChristophTobler/Firmware.git
+cd Firmware
+```
+
+Once you're in your local copy of the Firmware:
+```
+git checkout pr-raw_imu_snappy
+git submodule update --init --recursive
+export FC_ADDON=<location-of-extracted-flight-controller-addon>
+make clean
+make eagle_default
+make eagle_default upload
+adb push ROMFS/px4fmu_common/mixers/quad_x.main.mix  /usr/share/data/adsp
+```
+
+## Install ROS
+Set up your snapdragon to connect to your local Wi-Fi network. To do so, open an adb shell and edit the file `/etc/wpa_supplicant/wpa_supplicant.conf` and add your local network settings:
```

Comment:
"snapdragon" to "Snapdragon". Generally good to capitalise proper names.

I'm not too concerned about using product names precisely (e.g. saying "Qualcomm Snapdragon Flight" every time you use it) but you might find that the Qualcomm marketing department are. 

Generally I use italic to mark up tool names. So *adb* shell.